### PR TITLE
[WICKET-7030] Add Convenience Methods to BaseWicketTester

### PR DESCRIPTION
**Pull request by renoth**
- Add getAllComponentsFromLastRenderedPageByWicketId and getFirstComponentFromLastRenderedPageByWicketId to BaseWicketTester

It has always bothered me that unlike TagTester BaseWicketTester has no way of finding Components by Wicket ID

### CODE REVIEW

File: wicket-core/src/main/java/org/apache/wicket/util/tester/BaseWicketTester.java [hunk 1]
```diff
@@ -1576,6 +1577,48 @@ public Component getComponentFromLastRenderedPage(String path)
 		return getComponentFromLastRenderedPage(path, true);
 	}
 
+	/**
+	 * TODO
```
Comment (renoth) [inline wicket-core/src/main/java/org/apache/wicket/util/tester/BaseWicketTester.java:1581]: still todo

File: wicket-spring/src/test/java/org/apache/wicket/spring/injection/bytebuddy/ParallelInjectionTest.java [hunk 2]
```diff
@@ -110,7 +110,7 @@ void notParallel() throws Exception {
     }
```
Comment (renoth) [inline wicket-spring/src/test/java/org/apache/wicket/spring/injection/bytebuddy/ParallelInjectionTest.java:110]: cleaned up some leftovers that are from my org

**Review by martin-g — APPROVED** [review]
We will need a JIRA ticket for the changelog.

File: wicket-core/src/main/java/org/apache/wicket/util/tester/BaseWicketTester.java [hunk 3]
```diff
@@ -1576,6 +1577,48 @@ public Component getComponentFromLastRenderedPage(String path)
...
+	 *
+	 * @param wicketId
+	 * @return
+	 */
+	public Optional<Component> getFirstComponentFromLastRenderedPageByWicketId(String wicketId) {
```
[the remark this answers is not in the document]
Comment (renoth) [reply wicket-core/src/main/java/org/apache/wicket/util/tester/BaseWicketTester.java]: done

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thank you, @renoth !

**Pull request merged**